lib/srv/db/ca.go

```diff
@@ -172,22 +183,107 @@ func (s *Server) getCACertPaths(database types.Database) ([]string, error) {
 	return nil, trace.BadParameter("%v doesn't support automatic CA download", database)
 }
 
+// saveCACert saves the downloaded certificate to the filesystem.
+func (s *Server) saveCACert(filePath string, content []byte, version []byte) error {
+	// Save CA contents.
+	err := os.WriteFile(filePath, content, teleport.FileMaskOwnerOnly)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Save the CA version.
+	err = os.WriteFile(filePath+versionFileSuffix, version, teleport.FileMaskOwnerOnly)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	s.log.Debugf("Saved CA certificate %v.", filePath)
+	return nil
+}
+
+// updateCACert updates the database CA contents if it has changed.
+func (s *Server) updateCACert(ctx context.Context, database types.Database, filePath string) error {
+	var contents []byte
+
+	// Get the current CA version.
+	version, err := s.cfg.CADownloader.GetVersion(ctx, database, filePath)
+	// If getting the CA version is not supported, download it.
+	if trace.IsNotImplemented(err) {
+		contents, version, err = s.cfg.CADownloader.Download(ctx, database, filePath)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	} else if err != nil {
+		return trace.Wrap(err)
+	}
+
+	equal, err := isCAVersionEqual(filePath, version)
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
+
+	if equal {
+		return nil
```

Comment:
nit: it would be great we can have a few more debug log in this function like here for easier debugging.